addons/udes_stock/models/stock_move_line.py

```diff
@@ -164,135 +167,145 @@ def mark_as_done(
-                    ("product_id", "in", mls_done.mapped("product_id").ids),
-                ]
-            )
-            self._trigger_print_for_move_line_mark_as_done(
-                active_model=picking._name,
-                active_ids=picking.ids,
-                print_records=print_mls,
-                action_filter="move_lines.mark_as_done",
-            )
+            mls_done = MoveLine.browse()
+            for ml in move_lines:
+                ml_values = values.copy()
+                # Check if there is specific info for the move_line product
+                # otherwise we fully mark as done the move_line
+                if products_info_by_product:
+                    # check if the qty done has been fulfilled
+                    if products_info_by_product[ml.product_id]["qty"] == 0:
+                        continue
+                    ml_values, products_info_by_product = ml._prepare_line_product_info(
+                        ml_values, products_info_by_product
+                    )
+                else:
+                    ml_values["qty_done"] = ml.product_qty
+                mls_done |= ml._mark_as_done(ml_values)
+
+            if loc_dest_instance is not None:
+                # HERE(ale): updating the dest loc here to have a single
+                # invocation of its constraint handler (see below)
+                mls_done.write({"location_dest_id": loc_dest_instance.id})
+
+            # TODO: at this point products_info_by_product should be with qty_todo = 0?
+            #       No necessarily, can we have add unexpected parts and not enough stock?
+
+            # it might be useful when extending the method
+            if parent_package:
+                mls_done.write({"u_result_parent_package_id": parent_package.id})
+
+            if result_package and picking is not None:
+                # Print the package label
+                self._trigger_print_for_pack_mark_as_done(
+                    active_model=picking._name,
+                    active_ids=picking.ids,
+                    print_records=result_package,
+                    action_filter="move_lines.mark_as_done",
+                )
 
+            if mls_done and picking is not None:
+                # Print the move line label
+                print_mls = MoveLine.search(
+                    [
+                        ("result_package_id", "in", mls_done.mapped("result_package_id").ids),
+                        ("product_id", "in", mls_done.mapped("product_id").ids),
+                    ]
+                )
+                self._trigger_print_for_move_line_mark_as_done(
+                    active_model=picking._name,
+                    active_ids=picking.ids,
+                    print_records=print_mls,
+                    action_filter="move_lines.mark_as_done",
+                )
+        # Write the log
+        _logger.info(
+            "(%s) %s (%s) mark_as_done in %.2fs, %d queries, %.2f q/s",
```

Comment:
It might be worth abstracting this message format, as it's common? ('mark_as_done' could become a `%s`)

Reply:
We could make a module and abstract these calls or do other clever things but, I didn't go further than what I've done right now to try and limit the work otherwise there's always something to improve. Happy to do more work as long as TL agrees.

Reply:
I think it would be helpful to prevent the formats from accidentally diverging, but happy for someone else to make the final call.

Reply:
@ws-mps any thoughts?

Reply:
It is a good idea to go to that direction, but it is out of scope of this story to create a new module etc. If we want to predefine somewhere the format (constant?) where we 'mark_as_done' (and others) become a %s, I'm fine with that.

Reply:
I will abstract and store the message format somewhere in udes_stock so that it is easier to change and track and then import it where it's used. How does it sound? @kdwyer